conans/client/installer.py

```diff
@@ -120,14 +122,41 @@ def _prepare_sources(self, conanfile, pref, package_layout, conanfile_path, sour
             logger.debug("BUILD: Copied to %s", build_folder)
             logger.debug("BUILD: Files copied %s", ",".join(os.listdir(build_folder)))
 
-    def _build(self, conanfile, pref):
+    def _build(self, conanfile, pref, node):
         # Read generators from conanfile and generate the needed files
         logger.info("GENERATORS: Writing generators")
         write_generators(conanfile, conanfile.build_folder, self._output)
 
         logger.info("TOOLCHAIN: Writing toolchain")
         write_toolchain(conanfile, conanfile.build_folder, self._output)
 
+        # I need the shims for the executables in the build-requires (only two-profiles approach)
+        logger.info("SHIMS: Writing shims for build-requires")
+        transitive = [it for it in node.transitive_closure.values()]
+        br_host = []
+        for it in node.dependencies:
+            if it.require.build_require_context == CONTEXT_HOST:
+                br_host.extend(it.dst.transitive_closure.values())
+
+        add_path = False
+        node_order = [n for n in node.public_closure if n.binary != BINARY_SKIP]
+        for n in node_order:
+            if n not in transitive and n not in br_host:
+                # FIXME: I can't use the ShimGenerator because it expects the consumer of the conanfiles
+                self._output.info("Shims for {}".format(n.conanfile))
```

Comment:
I would expect this to share the same code as the generators.

Reply:
I need to double-check this, but I think this is one of the problems we have right now with generators. Generators are conceived to generate files for the requirements of a conanfile, but not for a conanfile itself.... maybe here I have a VIRTUAL conanfile and I can use it to feed the generator. I will check